test/e2e/must-gather/must_gather_suite_test.go

```diff
@@ -62,12 +62,13 @@ var _ = ginkgo.BeforeSuite(func() {
 	mustGatherTag = getStringValueFromEnv(envVarMustGatherTag, defaultMustGatherTag)
 	ginkgo.By(fmt.Sprintf("Using must-gather image %q tag %q", mustGatherImage, mustGatherTag))
 
+	ginkgo.By("Fetching up cluster data")
+	var err error
+	// the error might be not nil we'll decide if that's fine or not depending on E2E_NROP_INFRA_SETUP_SKIP
+	deployment, err = deploy.GetDeploymentWithSched(context.TODO())
```

Comment:
this should never fail anyway so let's check err not occurred

Reply:
it can fail in case the cluster doesn't have the scheduler and we try to fetch it

Reply:
I think the question is why the sudden change? why are we always running the deployment, even if `E2E_NROP_INFRA_SETUP_SKIP` exists.

Reply:
my take is that we always want to have an instance of the current scheduler object. If the scheduler does already exist (skip setup), then we should just fetch the existing instance and move on; otherwise we need to setup it and then fetch the setupped instance. This is probably not too evident from the code and we likely need a bit larger refactoring.